### PR TITLE
added  socket 4/5 support!

### PR DESCRIPTION
**Pull request by tailot**

### CODE REVIEW

**Comment by soyuka** [comment]
I really want to improve the options exactly for this we're missing a bunch and it's not so great to edit so many files just to add one. 

@cazala You can merge this if you agree with the changes (I do) but we should wait a couple of days before issuing a new release (because at this rythm binary arguments will be a mess :P).

**Comment by tailot** [comment]
@soyuka Good observation. I suggest a good parser https://github.com/substack/minimist

**Comment by soyuka** [comment]
Exactly, also I want to refactor the constructor arguments to an object see #7. 

Just need time to do so and dunno if I'll do it tonight haha.

**Comment by cazala** [comment]
Hi @tailot, after merging #11 this has conflicts 👇

**Comment by tailot** [comment]
H @cazala, done!

File: bin/coin-hive [hunk 1]
```diff
@@ -31,7 +31,8 @@ if (argv.help || argv.h) {
     interval: argv.interval || process.env.COINHIVE_INTERVAL || defaults.interval ,
     port: argv.port || process.env.COINHIVE_PORT || defaults.port,
     host: argv.host || process.env.COINHIVE_HOST || defaults.host ,
-    threads: argv.threads || process.env.COINHIVE_THREADS || defaults.threads
+    threads: argv.threads || process.env.COINHIVE_THREADS || defaults.threads,
+    proxy: argv.proxy
```
Comment (soyuka) [inline bin/coin-hive]: what about configuring this through env variables?

Comment (cazala) [reply bin/coin-hive]: +1

**Pull request merged**